test/spec.yml

```diff
@@ -1333,3 +1334,213 @@
+  steps:
+    - in: bn trigger setup $TRIGGER_NAME --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io#teststream2" --target $FUNC_NAME
+      out: UPDATED
+
+- test: Test 2 new triggers, same source, different target (good-path)
+  setup:
+    - export FUNC_NAME=rand$RANDOM$RANDOM
+    - export FUNC_NAME2=rand$RANDOM$RANDOM
+    - export TRIGGER_NAME=rand$RANDOM$RANDOM
+    - export TRIGGER_NAME2=rand$RANDOM$RANDOM
+    - bn create node8 $FUNC_NAME
+    - bn deploy $FUNC_NAME
+    - |-
+        cat <<EOF >/home/dockeruser/test/binaris.yml
+        functions:
+          $FUNC_NAME:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+          $FUNC_NAME2:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+        EOF
+    - bn deploy $FUNC_NAME2
+    - bn trigger setup $TRIGGER_NAME --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io#teststream" --target $FUNC_NAME
+  cleanup:
+    - bn trigger remove $TRIGGER_NAME
+    - bn trigger remove $TRIGGER_NAME2
+    - bn remove $FUNC_NAME
+    - bn remove $FUNC_NAME2
+  steps:
+    - in: bn trigger setup $TRIGGER_NAME2 --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io#teststream" --target $FUNC_NAME2
+      out: CREATED
+
+- test: Test new trigger with bad source uri (bad-path)
+  setup:
+    - export FUNC_NAME=rand$RANDOM$RANDOM
+    - export TRIGGER_NAME=rand$RANDOM$RANDOM
+    - bn create node8 $FUNC_NAME
+    - bn deploy $FUNC_NAME
+  cleanup:
+    - bn remove $FUNC_NAME
+  steps:
+    - in: bn trigger setup $TRIGGER_NAME --source-uri redis://not-a-valid-redis-uri --target $FUNC_NAME
+      exit: 1
+      err: |-
+        RequestId: *
+        Could not connect to source*
+
+- test: Test new trigger with missing target function (bad-path)
+  steps:
+    - in: bn trigger setup notexistingfunctrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io#teststream" --target nonexistingfunc
+      exit: 1
+      err: |-
+        RequestId: *
+        Error: No such function
+
+- test: Test new trigger when it already exists with different name (bad-path)
+  setup:
+    - export FUNC_NAME=rand$RANDOM$RANDOM
+    - export TRIGGER_NAME=rand$RANDOM$RANDOM
+    - bn create node8 $FUNC_NAME
+    - bn deploy $FUNC_NAME
+    - bn trigger setup $TRIGGER_NAME --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io#teststream" --target $FUNC_NAME
+  cleanup:
+    - bn trigger remove $TRIGGER_NAME
+    - bn remove $FUNC_NAME
+  steps:
+    - in: bn trigger setup testduplicatetriggerthatwillnotbecreated --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io#teststream" --target $FUNC_NAME
+      exit: 1
+      err: |-
+        RequestId: *
+        Bad Request, errors: Trigger from source to target already exists
+
+- test: Test update trigger when it already exists with different name (bad-path)
```

Comment:
Come to think about this again, this test is redundant, it's already tested in the test above.